src/server/snapshot.h

```diff
@@ -13,6 +15,25 @@
 
 namespace dfly {
 
+// TODO: Change to virtual interface?
+struct SnapshotSyncBlock {
+  explicit SnapshotSyncBlock(unsigned threads, std::atomic_uint16_t* counter)
+      : barrier_(threads), counter_(counter) {
+  }
+
+  void wait() {
+    counter_->fetch_add(-1, std::memory_order_relaxed);
+    barrier_.wait();
+  }
+
+ private:
+  // Barrier to sync consumer fibers on CONTINIOUS_JOURNAL mode.
+  // TODO: No wait until. No fail behaviour.
+  ::boost::fibers::barrier barrier_;
+  // Atomic flag
+  std::atomic_uint16_t* counter_;
+};
+
```

Comment:
why do you need a barrier? i.e. why do you need to block in snapshot (producer side) ? I would assume that notification is enough

Reply:
It notifies it inside the flow (see `SendFullSyncCut`).

However we have to stop the code flow somehow, which proceeds with sending EOF and quitting. This can be done by either:
- splitting the process into two fibers
- blocking the fibers (journal callbacks fire nonetheless)

What other solution can you suggest? Change the saver completely to have a two-step interface?